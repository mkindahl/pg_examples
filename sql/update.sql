CREATE SCHEMA IF NOT EXISTS simple_catalog;
CREATE EXTENSION simple WITH VERSION '1.0' SCHEMA simple_catalog;

CREATE ROLE reader;

GRANT SELECT ON ALL TABLES IN SCHEMA simple_catalog TO reader;
ALTER DEFAULT PRIVILEGES IN SCHEMA simple_catalog GRANT SELECT ON TABLES TO reader;

SELECT nspname, defaclacl FROM pg_namespace ns JOIN pg_default_acl ON ns.oid = defaclnamespace;

SELECT nspname, relname, relacl, privtype, initprivs
FROM pg_class cl JOIN pg_namespace ns ON ns.oid = relnamespace LEFT JOIN pg_init_privs ON objoid = cl.oid
WHERE relname = 'objects';

SELECT nspname, nspacl, privtype, initprivs
FROM pg_namespace ns LEFT JOIN pg_init_privs ON objoid = ns.oid
WHERE nspname = 'simple_catalog';

CREATE TABLE update_foo (a int);
CALL add_note('update_foo', 'Just a note');
SELECT * FROM simple_catalog.objects;
ALTER EXTENSION simple UPDATE;
SELECT * FROM simple_catalog.objects;

SELECT nspname, nspacl, privtype, initprivs
FROM pg_namespace ns LEFT JOIN pg_init_privs ON objoid = ns.oid
WHERE nspname = 'simple_catalog';

SELECT nspname, relname, relacl, privtype, initprivs
FROM pg_class cl JOIN pg_namespace ns ON ns.oid = relnamespace LEFT JOIN pg_init_privs ON objoid = cl.oid
WHERE relname = 'objects';

DROP EXTENSION simple;
DROP SCHEMA simple_catalog;
DROP TABLE update_foo;
DROP ROLE reader;
