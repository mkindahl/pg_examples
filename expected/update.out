CREATE SCHEMA IF NOT EXISTS simple_catalog;
CREATE EXTENSION simple WITH VERSION '1.0' SCHEMA simple_catalog;
CREATE ROLE reader;
GRANT SELECT ON ALL TABLES IN SCHEMA simple_catalog TO reader;
ALTER DEFAULT PRIVILEGES IN SCHEMA simple_catalog GRANT SELECT ON TABLES TO reader;
SELECT nspname, defaclacl FROM pg_namespace ns JOIN pg_default_acl ON ns.oid = defaclnamespace;
    nspname     |    defaclacl    
----------------+-----------------
 simple_catalog | {reader=r/mats}
(1 row)

SELECT nspname, relname, relacl, privtype, initprivs
FROM pg_class cl JOIN pg_namespace ns ON ns.oid = relnamespace LEFT JOIN pg_init_privs ON objoid = cl.oid
WHERE relname = 'objects';
    nspname     | relname |                  relacl                   | privtype |          initprivs          
----------------+---------+-------------------------------------------+----------+-----------------------------
 simple_catalog | objects | {mats=arwdDxt/mats,=r/mats,reader=r/mats} | e        | {mats=arwdDxt/mats,=r/mats}
(1 row)

SELECT nspname, nspacl, privtype, initprivs
FROM pg_namespace ns LEFT JOIN pg_init_privs ON objoid = ns.oid
WHERE nspname = 'simple_catalog';
    nspname     | nspacl | privtype | initprivs 
----------------+--------+----------+-----------
 simple_catalog |        |          | 
(1 row)

CREATE TABLE update_foo (a int);
CALL add_note('update_foo', 'Just a note');
SELECT * FROM simple_catalog.objects;
   objoid   |   objnote   
------------+-------------
 update_foo | Just a note
(1 row)

ALTER EXTENSION simple UPDATE;
SELECT * FROM simple_catalog.objects;
   objoid   |         objnote          
------------+--------------------------
 update_foo | {"note" : "Just a note"}
(1 row)

SELECT nspname, nspacl, privtype, initprivs
FROM pg_namespace ns LEFT JOIN pg_init_privs ON objoid = ns.oid
WHERE nspname = 'simple_catalog';
    nspname     | nspacl | privtype | initprivs 
----------------+--------+----------+-----------
 simple_catalog |        |          | 
(1 row)

SELECT nspname, relname, relacl, privtype, initprivs
FROM pg_class cl JOIN pg_namespace ns ON ns.oid = relnamespace LEFT JOIN pg_init_privs ON objoid = cl.oid
WHERE relname = 'objects';
    nspname     | relname |                  relacl                   | privtype |                 initprivs                 
----------------+---------+-------------------------------------------+----------+-------------------------------------------
 simple_catalog | objects | {mats=arwdDxt/mats,reader=r/mats,=r/mats} | e        | {mats=arwdDxt/mats,reader=r/mats,=r/mats}
(1 row)

DROP EXTENSION simple;
DROP SCHEMA simple_catalog;
DROP TABLE update_foo;
DROP ROLE reader;
